src/EnergyPlus/Material.cc

```diff
@@ -2884,4 +2871,221 @@ void GetVariableAbsorptanceInput(EnergyPlusData &state, bool &errorsFound)
     }
 }
 
+void CalcScreenTransmittance(EnergyPlusData &state,
+                             MaterialScreen const *screen,
+                             Real64 phi,     // Sun altitude relative to surface outward normal (radians, 0 to Pi)
+                             Real64 theta,   // Optional sun azimuth relative to surface outward normal (radians, 0 to Pi)
+                             ScreenBmTransAbsRef &tar
+)
+{
+
+    // FUNCTION INFORMATION:
+    //       AUTHOR         Richard Raustad
+    //       DATE WRITTEN   May 2006
+    //       MODIFIED       na
+    //       RE-ENGINEERED  na
+
+    // PURPOSE OF THIS FUNCTION:
+    //  Calculate transmittance of window screen given azimuth and altitude angle
+    //  of sun and surface orientation.
+
+    // METHODOLOGY EMPLOYED:
+    //  Window screen solar beam transmittance varies as the sun moves across the sky
+    //  due to the geometry of the screen material and the angle of incidence
+    //  of the solar beam. Azimuth and altitude angle are calculated with respect
+    //  to the surface outward normal. Solar beam reflectance and absorptance are also
+    //  accounted for.
+
+    Real64 constexpr Small(1.E-9); // Small Number used to approximate zero
+
+    Real64 Tdirect;                   // Beam solar transmitted through screen (dependent on sun angle)
+    Real64 Tscattered;                // Beam solar reflected through screen (dependent on sun angle)
+    Real64 TscatteredVis;             // Visible beam solar reflected through screen (dependent on sun angle)
+    Real64 Tscattermax;               // Maximum solar beam  scattered transmittance
+    Real64 TscattermaxVis;            // Maximum visible beam scattered transmittance
+    Real64 ExponentInterior;          // Exponent used in scattered transmittance calculation
+    // when Delta < DeltaMax (0,0 to peak)
+    Real64 ExponentExterior; // Exponent used in scattered transmittance calculation
+    // when Delta > DeltaMax (peak to max)
+
+    assert(phi >= 0.0 && phi <= Constant::Pi);
+    assert(theta >= 0.0 && theta <= Constant::Pi);
+    
+    Real64 sinPhi = std::sin(phi);
+    Real64 cosPhi = std::cos(phi);
+    Real64 tanPhi = sinPhi / cosPhi;
+    Real64 sinTheta = std::sin(theta);
+    Real64 cosTheta = std::cos(theta);
+    Real64 tanTheta = sinTheta / cosTheta;
+    
+    bool sunInFront = (phi < Constant::PiOvr2) && (theta < Constant::PiOvr2); // Sun is in front of screen
```

Comment:
If `phi` and `theta` are both less than `Pi/2` then the sun is "in front" of the screen, i.e., the screen is between the sun and the window.